substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrThreadLocal.java

```diff
@@ -82,16 +83,21 @@ public void initialize(long bufferSize) {
     @Uninterruptible(reason = "Accesses a JFR buffer.")
     @Override
     public void beforeThreadRun(IsolateThread isolateThread, Thread javaThread) {
-        // We copy the thread id to a thread-local in the IsolateThread. This is necessary so that
-        // we are always able access that value without having to go through a heap-allocated Java
-        // object.
-        Target_java_lang_Thread t = SubstrateUtil.cast(javaThread, Target_java_lang_Thread.class);
-        traceId.set(isolateThread, t.getId());
+        // We copy the parent thread id to a thread-local in the IsolateThread. This is necessary so
+        // that we are always able to access that value without having to go through a
+        // heap-allocated Java object. Also, we are adding +1 for error checks.
+        parentThreadId.set(isolateThread, JavaThreads.getParentThreadId(javaThread) + 1);
```

Comment:
`parentThreadId` can be 0, so I was adding +1/-1 to keep 0 as an error check, but after reverting changes with `traceId` this will become invalid as well.